### ISSUE TITLE
Num of speakers

### ISSUE THREAD

**Issue by asr-lord** [philipperemy/deep-speaker#88]
What is the number of speakers used in ResCNN Softmax model? 921 or 2484?

![image](https://user-images.githubusercontent.com/83761777/128182266-dfdea152-b461-4132-9ba9-c81bc85ffa51.png)

![image](https://user-images.githubusercontent.com/83761777/128182311-09cb81bf-f86a-4a4f-86ed-0adfc42bad44.png)

**Comment by philipperemy** [comment]
@asr-lord this is the number of speakers used in the pre-training with softmax. It's technically possible to train on more speakers with softmax but the philosophy of the paper was to pre-train the model first on m<N where N is the final number of speakers in the dataset. The main training (softmax+triplet training) uses all of them.

**Comment by asr-lord** [comment]
It's clear now, thank you.

**Comment by lhang33** [comment]
I have tested the pre-training and the triplet-training this week. In my test, The pre-training part have create a BIG object named "KerasFormatConverter" which takes around 32GB RAM and 10GB GPU-RAM when training on 923 speakers. And when I train on  smaller numbers of speakers, the RAM usage drops quickly (which I think because of the softmax layer in the pretrain stage).
So, I think the paper use a less speaker dataset to do the pre-training is because of the hardware limits. If we who have better hardware, do the pre-training stage on the whole dataset will be better!
